plugins/dev-tools/src/common_test_helpers.mocha.js

```diff
@@ -6,26 +6,76 @@
 
 /**
  * Test case configuration information.
- * @typedef {{
- *            title:string,
- *            value:*
- *            skip:(boolean|undefined),
- *            only:(boolean|undefined),
- *            expectedValue:?
- *          }}
- */
-export let TestCase;
+ * @record
+ */
+export function TestCase() {}
+/**
+ * @type {string} The title for the test case.
+ */
+TestCase.prototype.title = '';
+/**
+ * @type {boolean|undefined} Whether this test case should be skipped. Used to
+ *    skip buggy test case and should have an associated bug.
+ */
+TestCase.prototype.skip = false;
+/**
+ * @type {boolean|undefined} Whether this test case should be called as only.
+ *    Used for debugging.
+ */
+TestCase.prototype.only = false;
+
+/**
+ * Test suite configuration information.
+ * @record
+ * @template {TestCase} T
+ */
+export function TestSuite() {}
+/**
+ * @type {string} The title for the test case.
+ */
+TestSuite.prototype.title = '';
+/**
+ * @type {boolean} Whether this test suite should be skipped. Used to
+ *    skip buggy test case and should have an associated bug.
+ */
+TestSuite.prototype.skip = false;
+/**
+ * @type {boolean} Whether this test suite should be called as only.
+ *    Used for debugging.s
+ */
+TestSuite.prototype.only = false;
+/**
+ * @type {Array<T>} The associated test cases.
+ */
+TestSuite.prototype.testCases = [];
 
 /**
  * Runs provided test cases.
- * @param {Array<TestCase>} testCases The test cases to run.
- * @param {function(TestCase):function} testFn Function that returns test
- *  callback.
+ * @param {Array<T>} testCases The test cases to run.
+ * @param {function(T):Function} createTestCallback Creates test
+ *    callback using given test case
+ * @template {TestCase} T
  */
-export function runTestCases(testCases, testFn) {
-  testCases.forEach((testCase) => {
+export function runTestCases(testCases, createTestCallback) {
+  testCases.forEach((/** @type {TestCase} */testCase) => {
     let testCall = (testCase.skip ? test.skip : test);
     testCall = (testCase.only ? test.only : testCall);
-    testCall(testCase.title, testFn(testCase));
+    testCall(testCase.title, createTestCallback(testCase));
+  });
+}
+
+/**
+ * Runs provided test suite.
+ * @param {Array<TestSuite<T>>} testSuites The test suites to run.
+ * @param {function(TestSuite<T>):function(TestCase<T>):Function
```

Comment:
should be instead `{function(TestSuite<T>):function(T):Function}`

Reply:
Yeah I would add parentheses around that second function for clarity.